.circleci/publish.sh

```diff
@@ -85,6 +85,19 @@ elif [[ "$CIRCLE_BRANCH" =~ ^run-e2e-with-rc\/.* ]] || [[ "$CIRCLE_BRANCH" =~ ^r
   # create release commit and release tags
   npx lerna version --preid=rc.$(git rev-parse --short HEAD) --exact --conventional-prerelease --conventional-commits --yes --no-push --include-merged-tags --message "chore(release): Publish rc [ci skip]" $(echo $force_publish_local_args) --no-commit-hooks
 
+  internal_cli_package_json_path="../packages/amplify-cli/package.json"
+  npm_cli_package_json_path="../packages/amplify-cli-npm/package.json"
+
+  internal_cli_version=$(jq '.version' $internal_cli_package_json_path)
+  npm_cli_version=$(jq '.version' $npm_cli_package_json_path)
+
+  if [[ $internal_cli_version != $npm_cli_version ]]; then
+    echo "Versions did not match."
+    echo "amplify-cli version: $internal_cli_version"
+    echo "amplify-cli-npm version: $npm_cli_version"
+    exit 1
+  fi
+
```

Comment:
Does publish run in PR validation workflow ?

Reply:
Ok I see this here https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/13340/workflows/5b8e6c6d-0cf5-4cdf-a5bc-337b33d80cba/jobs/582609 .

My suggestion is to extract this code to separate function (and perhaps move to local_publish_helpers).
Then add explicit step to PR job that validates this. Adding this here makes it hard to discover and reason about for future readers.

Reply:
is the pathing correct? looks like its not finding the file:

![image](https://user-images.githubusercontent.com/110861985/206524822-e7091c8b-9b87-4c11-b144-00d05df8c72c.png)

Reply:
The path should be `./packages/amplify-cli/package.json`

Reply:
@sobolk I will add an explicit step in CircleCI